tests/tamr_client/fake.py

```diff
@@ -27,7 +47,28 @@ def _to_kwargs(fake):
         path = req.get("path")
         url = "http://localhost/api/versioned/v1/" + path
 
-    return dict(method=req["method"], url=url, **resp)
+    # Get response body from either ndjson or json
+    if resp.get("ndjson") is not None:
+        resp["body"] = "\n".join((dumps(line) for line in resp["ndjson"]))
+    elif resp.get("json") is not None:
+        resp["body"] = dumps(resp["json"])
+
+    # Get expected request body from ndjson
+    if req.get("ndjson") is not None:
+        req["body"] = [x for x in req["ndjson"]]
+    elif req.get("json") is not None:
+        req["body"] = req["json"]
+
+    rsps.add_callback(
+        method=req["method"],
+        url=url,
+        callback=partial(
+            check_request_body,
+            request_body=req.get("body"),
+            status=resp["status"],
```

Comment:
_Does_ every response need a status? The `responses` API suggests "no". What's the rationale for asserting this?

Reply:
The `callback` function parameter of `responses.add_callback` must return a tuple of [`(status, headers, body)`](https://github.com/getsentry/responses/blob/c21ba81b0053a887d412227acadb8d65b331db94/responses.py#L389-L395).  The implementation of `request.Response.raise_for_status()` requires this to be an `int`.  One way of handling this would be to pass `resp.get("status", 200)` (or some other default value), but this seems slightly problematic since it is causing the mock response to return a status that might not be reflective of the actual server response status.

I was under the impression that any real server response should return a status code, but I'm not basing that on much besides the above.  If we choose to use a default value, maybe `0` is reasonable, as that is what `urllib3` does [here](https://github.com/urllib3/urllib3/blob/f0c43419e45db17a3c5f287be80aa0d14e2f58c6/src/urllib3/response.py#L355).

Reply:
Ah ok. Weird that the status is optional for `responses.add`, but required for `responses.add_callback`. Makes sense.